Userland/Utilities/pro.cpp

```diff
@@ -179,6 +184,28 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
             request_headers.set(header.substring_view(0, split.value()), header.substring_view(split.value() + 1));
             return true;
         } });
+    args_parser.add_option(Core::ArgsParser::Option {
+        .argument_mode = Core::ArgsParser::OptionArgumentMode::Required,
+        .help_string = "(HTTP only) Provide basic authentication credentials (format: username:password)",
+        .long_name = "auth",
+        .short_name = 'u',
+        .value_name = "credentials",
+        .accept_value = [&](auto* s) {
+            StringView credentials { s, strlen(s) };
+            auto split = credentials.find(':');
```

Comment:
Actually there's no need for splitting the credentials into a `username:password` pair and then re-attaching them back together later. This also avoids a potential problem if you want to send both an empty username and empty password (There might be a server that 401's without an authorization header but accepts an empty username:password pair)

Reply:
Hm... I thought about that yesterday but then again: If they are split up, the parameter can be easily reused for other auth schemes that are not basic auth. So while it seems a little wasteful right now, it might be good to go for others.

I can gladly change it though, or offer the pure "plain credentials" as an additional variable that I'll reuse later instead of making my own. How does that sound?

Reply:
Sounds good to me

Reply:
Changed as discussed. I left splitting up until we actually need it, in pure Serenity fashion :^)